examples/lighting-app/nrfconnect/CMakeLists.txt

```diff
@@ -30,16 +30,36 @@ if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/boards/${BOARD}.conf)
   list(APPEND CONF_FILE boards/${BOARD}.conf)
 endif()
 
-option(BUILD_WITH_DFU "Build target with Device Firmware Upgrade support" OFF)
-if(BUILD_WITH_DFU)
+if(NOT BOARD STREQUAL "nrf52840dongle_nrf52840")
+    set(BUILD_WITH_DFU "MATTER" CACHE STRING "Build target with Device Firmware Upgrade support")
+else()
+    set(BUILD_WITH_DFU "OFF" CACHE STRING "Build target with Device Firmware Upgrade support")
+endif()
+
+if(NOT BUILD_WITH_DFU STREQUAL "OFF")
+    if(BUILD_WITH_DFU STREQUAL "BLE")
+        if(BOARD STREQUAL "nrf5340dk_nrf5340_cpuapp")
+            list(INSERT OVERLAY_CONFIG 0 ${CHIP_ROOT}/config/nrfconnect/app/overlay-multi_image_smp_dfu_support.conf)
+        elseif(BOARD STREQUAL "nrf52840dk_nrf52840")
+            list(INSERT OVERLAY_CONFIG 0 ${CHIP_ROOT}/config/nrfconnect/app/overlay-single_image_smp_dfu_support.conf)
+        endif()
+    elseif(NOT BUILD_WITH_DFU STREQUAL "MATTER")
+        message(FATAL_ERROR "Selected invalid BUILD_WITH_DFU value: ${BUILD_WITH_DFU}")
+    endif()
+
+    list(INSERT OVERLAY_CONFIG 0 ${CHIP_ROOT}/config/nrfconnect/app/overlay-mcuboot_qspi_nor_support.conf)
+    # Enable Matter OTA Requestor
     list(INSERT OVERLAY_CONFIG 0 ${CHIP_ROOT}/config/nrfconnect/app/overlay-ota_requestor.conf)
-    if(${BOARD} STREQUAL "nrf5340dk_nrf5340_cpuapp")
-        list(INSERT OVERLAY_CONFIG 0 ${CHIP_ROOT}/config/nrfconnect/app/overlay-multi_image_dfu_support.conf)
+
+    if(BOARD STREQUAL "nrf5340dk_nrf5340_cpuapp")
+        # DFU over Matter doesn't support multi-image update yet, but using this configs should not harm it anyway.
         set(mcuboot_OVERLAY_CONFIG ${CMAKE_CURRENT_SOURCE_DIR}/configuration/mcuboot_multi_image_dfu.conf CACHE INTERNAL "")
-    else()
-        list(INSERT OVERLAY_CONFIG 0 ${CHIP_ROOT}/config/nrfconnect/app/overlay-single_image_dfu_support.conf)
+    elseif(BOARD STREQUAL "nrf52840dk_nrf52840")
         set(mcuboot_OVERLAY_CONFIG ${CMAKE_CURRENT_SOURCE_DIR}/configuration/mcuboot_single_image_dfu.conf CACHE INTERNAL "")
+    else()
```

Comment:
Custom boards would need to extend/modify this CMakeLists.txt. Would it be possible to add the mcuboot overlay as part of the board's defconfig or have an option that let's a custom board pass the overlay without an FATAL_ERROR?

Reply:
Right, I removed this check. Basically I wanted only to disable building  target with DFU enabled for nRF52840dongle, but that is actually already done https://github.com/project-chip/connectedhomeip/pull/14144/files#diff-3de4731c08fd864006e9afb1b138e33ef6b01e7844a0f26abf875dec6c2718bcR36 so further checks were not needed